src/test/java/uk/gov/justice/hmpps/casenotes/services/ExternalApiServiceTest.kt

```diff
@@ -0,0 +1,510 @@
+@file:Suppress("ClassName")
+
+package uk.gov.justice.hmpps.casenotes.services
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.any
+import org.mockito.kotlin.check
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import org.springframework.data.domain.Sort
+import org.springframework.data.domain.Sort.Direction
+import org.springframework.data.domain.Sort.Order
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClient.RequestBodySpec
+import org.springframework.web.reactive.function.client.WebClient.RequestBodyUriSpec
+import org.springframework.web.reactive.function.client.WebClient.RequestHeadersSpec
+import org.springframework.web.reactive.function.client.WebClient.RequestHeadersUriSpec
+import org.springframework.web.reactive.function.client.WebClient.ResponseSpec
+import org.springframework.web.util.UriComponentsBuilder
+import reactor.core.publisher.Mono
+import uk.gov.justice.hmpps.casenotes.dto.BookingIdentifier
+import uk.gov.justice.hmpps.casenotes.dto.CaseNoteFilter
+import uk.gov.justice.hmpps.casenotes.dto.CaseNoteTypeDto
+import uk.gov.justice.hmpps.casenotes.dto.NewCaseNote
+import uk.gov.justice.hmpps.casenotes.dto.NomisCaseNote
+import uk.gov.justice.hmpps.casenotes.dto.OffenderBooking
+import uk.gov.justice.hmpps.casenotes.dto.UpdateCaseNote
+import java.time.LocalDateTime
+
+class ExternalApiServiceTest {
+  private val responseSpecMock: ResponseSpec = mock()
+  private val prisonApiWebClient: WebClient = mock()
+  private val authWebClient: WebClient = mock()
+  private val prisonApiClientCredentialsWebClient: WebClient = mock()
+  private val requestHeadersUriSpec: RequestHeadersUriSpec<*> = mock()
+  private val requestHeadersSpec: RequestHeadersSpec<*> = mock()
+  private val requestBodyUriSpec: RequestBodyUriSpec = mock()
+  private val requestBodySpec: RequestBodySpec = mock()
```

Comment:
Interesting approach. Asides from being much faster, are there are other pros/cons of not using Wiremock?

Reply:
Although there was lots of mocking at the beginning in the `setup()` after that it wasn't that bad.
Also allowed me fine grained testing of exactly what was going on.
Planning on using wiremock in the resource test, but this felt like it really should be a unit test instead